src/gui/torrentcontenttreeview.cpp

```diff
@@ -106,6 +108,29 @@ void TorrentContentTreeView::keyPressEvent(QKeyEvent *event)
     }
 }
 
+void TorrentContentTreeView::mouseDoubleClickEvent(QMouseEvent *event)
+{
+    const QPersistentModelIndex index = indexAt(event->pos());
+    if (index.isValid())
+    {
+        const QVariant checkValue = index.data(Qt::CheckStateRole);
+        if (checkValue.isValid() && !checkValue.isNull())
```

Comment:
Last question: isn't checking for null redundant?

Reply:
copied from here https://code.woboq.org/qt5/qtbase/src/widgets/itemviews/qstyleditemdelegate.cpp.html#297

Reply:
Ok.